diff --git a/Makefile.plain b/Makefile.plain
index aaff7ca..b622f85 100644
--- a/Makefile.plain
+++ b/Makefile.plain
@@ -24,7 +26,7 @@ install: install-check
 	@if [ "x$(DISABLE_GETTEXT)" = x ]; then (cd po && $(MAKE) -f Makefile.plain install); fi
 	@echo "Creating $(bindir)/jhbuild"
 	@mkdir -p $(bindir)
-	@sed -e "s,@jhbuilddir@,`pwd`,g" \
+	@sed -e "s,@jhbuilddir@,`pwd -W 2>&- || pwd`,g" \
 		-e "s,USE_CHECKOUT_SRC = False,USE_CHECKOUT_SRC = True," < \
 		scripts/jhbuild.in > $(bindir)/jhbuild
 	@chmod a+x $(bindir)/jhbuild
